app/helpers/local_time_helper.rb

```diff
@@ -54,51 +60,65 @@ def extract_options_and_value(options, value_key = nil)
       end
     end
 
-    def find_time_formats(format12)
-      if format12.is_a?(Symbol)
-        find_time_formats_by_name(format12)
-      else
-        [ format12.presence || LocalTime.default_time_format, nil ]
-      end
-    end
-
-    def find_time_formats_by_name(name)
-      if use_i18n_time_formats?(name)
-        find_i18_time_formats(name)
-      elsif use_ruby_time_formats?(name)
-        find_ruby_time_formats(name)
+    # This method makes more sense, but it makes a legacy test fail.
+    # def find_format_data(format, type)
+    #   data = if format.is_a?(Symbol)
+    #     use_i18n_format?(format) ? find_i18n_formats(format, type) : find_ruby_formats(format, type)
+    #   else
+    #     { format: format }
+    #   end
+    #   data[:format] ||= default_format(type)
+    #   data
+    # end
+
+    def find_format_data(format, type)
+      if format.is_a?(Symbol)
+        data = use_i18n_format?(format) ? find_i18n_formats(format, type) : find_ruby_formats(format, type)
+        data[:format] ||= LocalTime.default_time_format
+        data
+      elsif format
+        { format: format }
       else
-        [ LocalTime.default_time_format, nil ]
+        { format: default_format(type)}
       end
     end
 
-    def use_i18n_time_formats?(name)
+    def use_i18n_format?(name)
       i18n_time_or_date_format(name).present?
     end
 
     def i18n_time_or_date_format(name)
       I18n.t("time.formats.#{name}", default: [ :"date.formats.#{name}", "" ]).presence
     end
 
-    def find_i18_time_formats(name)
-      [ i18n_time_or_date_format(name), i18n_time_or_date_format("#{name}_24h") ]
-    end
-
-    def use_ruby_time_formats?(name)
-      ruby_time_or_date_format(name).present?
+    def find_i18n_formats(name, type)
+      if type == :time
+        { format: i18n_time_or_date_format(name), format24: i18n_time_or_date_format("#{name}_24h") }
+      else
+        { format: i18n_time_or_date_format(name) }
+      end
     end
 
-    def ruby_time_or_date_format(name)
-      Time::DATE_FORMATS.with_indifferent_access[name] || Date::DATE_FORMATS.with_indifferent_access[name]
+    def find_ruby_formats(name, type)
+      format_data = ruby_format(name, type)
+      if format_data.is_a?(Proc)
+        { format: default_format(type) }
+      elsif type == :time
+        { format: format_data, format24: ruby_format("#{name}_24h") }
+      else
+        { format: format_data }
```

Comment:
Returning formats as a hash is a nice idea. But I'd prefer if the method didn't have to know whether it needs to return two formats or just one. And since we've already gone with returning a "tuple" (array, really) I'd rather we stick with that.